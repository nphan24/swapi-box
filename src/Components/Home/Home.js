import React from 'react';
const homeImg = require('../../images/home-image.jpg');

const Home = () => {
  return (
    <div>
      <h1>Click on a button to see the various information!</h1>
      <img src={homeImg} alt='Star Wars logo' width='700px' height='500px'/>
    </div>
  );
};

export default Home;