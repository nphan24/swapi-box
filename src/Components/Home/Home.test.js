import React from 'react';
import Home from './Home';
import { shallow } from 'enzyme';

describe('Home', () => {
  let wrapper;
  
  beforeEach(() => {
    wrapper = shallow(<Home/>);
  });

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});