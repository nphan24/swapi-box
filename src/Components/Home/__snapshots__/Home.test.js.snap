// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Click on a button to see the various information!
</h1>,
        <img
          alt="Star Wars logo"
          height="500px"
          src="home-image.jpg"
          width="700px"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Click on a button to see the various information!",
        },
        "ref": null,
        "rendered": "Click on a button to see the various information!",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Star Wars logo",
          "height": "500px",
          "src": "home-image.jpg",
          "width": "700px",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Click on a button to see the various information!
</h1>,
          <img
            alt="Star Wars logo"
            height="500px"
            src="home-image.jpg"
            width="700px"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Click on a button to see the various information!",
          },
          "ref": null,
          "rendered": "Click on a button to see the various information!",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Star Wars logo",
            "height": "500px",
            "src": "home-image.jpg",
            "width": "700px",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
